src/SeqCli/Cli/Commands/Bench/BenchCommand.cs

```diff
@@ -0,0 +1,213 @@
+using Seq.Api.Model.Signals;
+using SeqCli.Cli.Features;
+using SeqCli.Connection;
+using Serilog;
+using Serilog.Context;
+using Serilog.Core;
+
+namespace SeqCli.Cli.Commands;
+
+/*
+ * Run performance benchmark tests against a Seq server.
+ *
+ * Requires test cases in a JSON file matching the format of `BenchCases.json`.
+ * 
+ * If a Seq reporting server is configured this command logs data such as:
+ * 
+ * {
+    "@t": "2022-11-09T01:12:06.0293545Z",
+    "@mt": "Bench run {BenchRunId} for query {Id}. Mean {MeanElapsed:N0} ms with relative dispersion {RelativeStandardDeviationElapsed:N2}",
+    "@m": "Bench run \"2907\" for query \"with-signal\". Mean 4 ms with relative dispersion 0.06",
+    "@i": "bb0c84a5",
+    "@r": [
+        "4",
+        "0.06"
+    ],
+    "BenchRunId": "2907",
+    "End": "2022-08-15T00:00:00.0000000",
+    "Id": "with-signal",
+    "LastResult": 606,
+    "MaxElapsed": 4.082,
+    "MeanElapsed": 3.6676,
+    "MinElapsed": 3.4334,
+    "Query": "select count(*) from stream where @Level = 'Warning'",
+    "RelativeStandardDeviationElapsed": 0.05619408341421253,
+    "Runs": 10,
+    "Signals": [
+        "signal-m33302"
+    ],
+    "Start": "2022-08-14T16:00:00.0000000"
+}
+ */
+[Command("bench", @"Measure query performance.
+
+Example cases file format:
+    
+{
+    ""cases"": [
+        {
+            ""id"": ""count-star"",
+            ""query"": ""select count(*) from stream"",
+            ""signals"": [""signal-id-here""]
+        }
+    ]
+}
+")]
+class BenchCommand : Command
+{
+    readonly SeqConnectionFactory _connectionFactory;
+    int _runs = 3;
+    readonly ConnectionFeature _connection;
+    readonly DateRangeFeature _range;
+    string _cases = "";
+    string _reportingServerUrl = "";
+    string _reportingServerApiKey = "";
+    ILogger? _reportingLogger = null;
+    
+    public BenchCommand(SeqConnectionFactory connectionFactory)
+    {
+        _connectionFactory = connectionFactory;
+        Options.Add("r|runs=", "The number of runs to execute", r =>
+        {
+            int.TryParse(r, out _runs);
+        });
+        
+        Options.Add(
+            "c|cases=", 
+            @"A JSON file containing the set of cases to run. Defaults to a standard set of cases.", 
+            c => _cases = c);
+
+        _connection = Enable<ConnectionFeature>();
+        _range = Enable<DateRangeFeature>();
+        
+        Options.Add(
+            "reporting-server=", 
+            "The address of a Seq server to send bench results to", 
+            s => _reportingServerUrl = s);
+        Options.Add(
+            "reporting-apikey=", 
+            "The API key to use when connecting to the reporting server", 
+            a => _reportingServerApiKey = a);
+    }
+    
+    protected override async Task<int> Run()
+    {
+        try
+        {
+            var connection = _connectionFactory.Connect(_connection);
+            _reportingLogger = BuildReportingLogger();
```

Comment:
Passing this around rather than storing it in a field will allow it to be cleaned up here via a `using` statement and ensure any buffered events are flushed.